hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/V1_31_1__Entity_Type_Mismatch.java

```diff
@@ -0,0 +1,332 @@
+    private List<EntityIdType> getEntityIdTypes(long entityId, long consensusTimestamp, int pageSize) throws SQLException {
+        log.info("Retrieve entityIdTypes for create transactions below entityId {} and before timestamp {} with page " +
+                "size {}", entityId, consensusTimestamp, pageSize);
+        List<EntityIdType> entityIdTypes = jdbcTemplate.query(
+                CREATED_ENTITIES_TRANSACTION_SQL,
+                new Object[] {entityId, consensusTimestamp, pageSize},
+                new RowMapper<>() {
+                    @Override
+                    public EntityIdType mapRow(ResultSet rs, int rowNum) throws SQLException {
+                        return getTypeMismatchedEntity(rs);
+                    }
+                });
+
+        if (entityIdTypes.isEmpty()) {
+            // no more rows to consider, return null
+            return null;
+        }
+
+        // remove nulls
+        entityIdTypes.removeAll(Collections.singleton(null));
+
+        log.debug("Retrieved {} entities with mismatch type from t_entities and transaction join", entityIdTypes
+                .size());
+        return entityIdTypes;
+    }
+
+    /**
+     * Batch update entities with correct fk_entity_type_id
+     *
+     * @param entityIdTypes List of mismatched entities
+     * @return
+     */
+    @Transactional
+    public int[][] batchUpdate(List<EntityIdType> entityIdTypes) {
+        log.trace("batchUpdate {} entities ", entityIdTypes.size());
+        return jdbcTemplate.batchUpdate(
+                ENTITIES_TYPE_UPDATE_SQL,
+                entityIdTypes,
+                flywayMigrationProperties.getEntityMismatchWriteBatchSize(),
+                new ParameterizedPreparedStatementSetter<>() {
+                    @Override
+                    public void setValues(PreparedStatement ps, EntityIdType entityIdType) throws SQLException {
+                        long id = entityIdType.entityId;
+                        ps.setLong(1, entityIdType.correctedEntityTypeId);
+                        ps.setLong(2, id);
+
+                        // update filter counters
+                        entityIdCap.set(id);
+                        timestampCap.set(entityIdType.consensusTimestamp);
+                    }
+                }
+        );
+    }
+
+    /**
+     * Retrieve the correct entityType number based on comparison between expected and current values When matched
+     * return 0 to signal equality, when mismatched return expectedType
+     *
+     * @param expectedEntityType
+     * @param currentEntityType
+     * @return
+     */
+    private int getCorrectedEntityType(EntityTypeEnum expectedEntityType, int currentEntityType) {
+        // check if EntityTypeEnum matches given currentEntityType.
+        // Return 0 on match otherwise return expected EntityTypeEnum id
+        if (expectedEntityType.getId() == currentEntityType) {
+            return 0;
+        }
+
+        return expectedEntityType.getId();
```

Comment:
Nit: not sure how you feel about ternary operators, but this method could be replaced with one I believe.
```suggestion
        return expectedEntityType.getId() == currentEntityType ? 0 :  expectedEntityType.getId();
```

Reply:
I like ternary operators and prefer your suggestion. Thanks